### PR TITLE
Example of aggregation + MPI splitting.

### PR DESCRIPTION
**Pull request by bdice**
@DomFijan asked about how to use aggregation with MPI, to split communicators. This PR shows an example of that workflow.

This workflow allows users to submit very large jobs with a single `mpirun` invocation that splits the communicator based on the number of jobs.

This PR requires aggregation in signac-flow and should not be merged until that feature is released: https://github.com/glotzerlab/signac-flow/pull/464

TODO:
- [x] Update requirements.txt to point to a new version of signac-flow that supports aggregation, once it is released.

### CODE REVIEW

File: projects/flow.aggregation-mpi/project.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+@Project.operation
+@directives(nranks=lambda *jobs: RANKS_PER_JOB * len(jobs))
+@aggregator.groupsof(JOBS_PER_AGGREGATE)
+def do_mpi_task(*jobs):
+    from mpi4py import MPI
+
+    comm = MPI.COMM_WORLD
+    rank = comm.Get_rank()
+    size = comm.Get_size()
```
Comment (joaander) [inline projects/flow.aggregation-mpi/project.py:20]:
```suggestion
    world_comm = MPI.COMM_WORLD
    world_rank = comm.Get_rank()
    world_size = comm.Get_size()
```
For symmetry and to disambiguate with `split_comm`, `split_rank`, and `split_size` below.

File: projects/flow.aggregation-mpi/project.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    num_jobs = len(jobs)
+    print(f"Launching MPI tasks for {len(jobs)} jobs. Rank {rank} of {size}.")
```
Comment (joaander) [inline projects/flow.aggregation-mpi/project.py:22]:
```suggestion
    if world_rank == 0:
        print(f"Launching MPI tasks for {len(jobs)} jobs. Rank {rank} of {size}.")
```
Print this message only once.

File: projects/flow.aggregation-mpi/project.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    # Use MPI splitting to make new communicators
+    communicator_size = size // num_jobs
+    color = rank // communicator_size
+    key = rank % communicator_size
+    print(f"{rank=}, {color=}, {key=}")
+
+    split_comm = comm.Split(color, key)
+    split_rank = split_comm.Get_rank()
+    split_size = split_comm.Get_size()
+    job = jobs[color]
```
Comment (joaander) [inline projects/flow.aggregation-mpi/project.py:33]:
```suggestion

    # Select the job from the aggregate to run in the split communicator.
    job = jobs[color]
```
This is the key step, it needs a little more explanation.

**Review by joaander — COMMENTED** [review]
This is excellent, thanks!

I have a few stylistic suggestions.

**Review by SchoeniPhlippsn — APPROVED** [review]
Sorry for the very late response. 
But I think this is a great example for splitting communicators and I don't have any suggestions for changes.

File: projects/flow.aggregation-mpi/project.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    pass
+
+
+RANKS_PER_JOB = 2
+JOBS_PER_AGGREGATE = 8
+
+
+@Project.operation
+@directives(nranks=lambda *jobs: RANKS_PER_JOB * len(jobs))
+@aggregator.groupsof(JOBS_PER_AGGREGATE)
```
Comment (atravitz) [inline projects/flow.aggregation-mpi/project.py:14]:
```suggestion
@aggregator.groupsof(num=JOBS_PER_AGGREGATE)
```

File: projects/flow.aggregation-mpi/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,2 @@
+signac>=1.0.0
+signac-flow>=0.14.0
```
Comment (atravitz) [inline projects/flow.aggregation-mpi/requirements.txt:2]: include mpi4py requirement here

**Review by atravitz — COMMENTED** [review]
Minor edits, but I also think it would be helpful and more complete to include a small function, rather than just the comment at the end.

**Comment by bdice** [comment]
@atravitz Take another look - I think I fixed the issues you noted. If you approve, please go ahead and merge. 👍

**Review by atravitz — APPROVED** [review]
LGTM!

**Pull request merged**